pkg/resources/mutation/cluster/provisioningcluster.go

```diff
@@ -328,9 +339,14 @@ func dropMachineSelectorFile(file rkev1.RKEProvisioningFiles, cluster *v1.Cluste
 }
 
 // MachineSelectorFileExists returns a boolean to indicate if the provided RKEProvisioningFiles exist in the provided cluster.
-func MachineSelectorFileExists(file rkev1.RKEProvisioningFiles, cluster *v1.Cluster) bool {
+func MachineSelectorFileExists(file rkev1.RKEProvisioningFiles, cluster *v1.Cluster, ignoreValueCheck bool) bool {
 	for _, item := range cluster.Spec.RKEConfig.MachineSelectorFiles {
-		if equality.Semantic.DeepEqual(file, item) {
+		fromCluster := item.DeepCopy()
+		if ignoreValueCheck {
+			cleanupHash(fromCluster)
```

Comment:
`DeepCopy()` is expensive so let's not call it unless we need to modify the object.
```suggestion
		fromCluster := item
		if ignoreValueCheck {
		    fromCluster = item.DeepCopy()
			cleanupHash(fromCluster)
```

Reply:
fixed.